### PR TITLE
feat: add parallel publisher mode

### PR DESCRIPTION
**Pull request by jakeloo**
## Summary
- replace old publisher modes with `default` and new `parallel` mode
- run commit and publish loops concurrently when in `parallel` mode and track last published blocks
- only publish during commits when in `default` mode
- initialize parallel publisher cursor from main storage to prevent re-publishing

## Testing
- `go test ./...` *(command produced no output)*

------
https://chatgpt.com/codex/tasks/task_b_6896dc37689c8332b734b4405875abf1

## Summary by CodeRabbit

* **New Features**
  * Introduced a configurable "publisher mode" with options for "default" (sequential) or "parallel" (concurrent commit and publish) operation, selectable via command-line flag or configuration file.
  * Added new configuration options and examples for publisher mode.
  * Added concurrent commit and publish processing to improve performance in parallel mode.
* **Bug Fixes**
  * Improved handling and tracking of published block numbers in storage backends.
* **Tests**
  * Added comprehensive tests for the new parallel publisher mode and related edge cases.
* **Chores**
  * Extended storage interfaces and mocks to support new publisher tracking functionality.

### CODE REVIEW

File: internal/orchestrator/committer.go [hunk 1]
```diff
@@ -79,15 +84,63 @@ func (c *Committer) Start(ctx context.Context) {
...
+			c.runCommitLoop(ctx, interval)
+		}()
+		<-ctx.Done()
+		wg.Wait()
+		log.Info().Msg("Committer shutting down")
+		c.publisher.Close()
+		return
+	}
+
+	c.runCommitLoop(ctx, interval)
```
Comment (nischitpra) [inline internal/orchestrator/committer.go:135]: this should be in the else if not parallel else it'll run twice
